### PR TITLE
Add preview package

### PR DESCRIPTION
**Pull request by TravisEz13**
Implements https://github.com/PowerShell/PowerShell-RFC/pull/134
@tylerl0706 Can your review?

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+version: 1.0.{build}
+environment:
+  matrix:
+  - APPVEYOR_BUILD_WORKER_IMAGE: ubuntu
+    Purpose: preview
+  - APPVEYOR_BUILD_WORKER_IMAGE: ubuntu
+    Purpose: stable
+
+install:
+#  - ps: Install-module pester -Scope CurrentUser -Force -SkipPublisherCheck
```
Comment (TylerLeonhardt) [inline appveyor.yml]: nit: any reason for this commented out code?

File: preview/snap/snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+version-script: |
+  file="./version.txt"
+  if [ -f "$file" ]
+  then
+    cat $file
+  else
+    version=$(curl -s 'https://raw.githubusercontent.com/PowerShell/PowerShell/master/tools/metadata.json' | jq .NextReleaseTag | sed 's/"//g' | sed 's/v//')
+    echo $version
+  fi
+summary: PowerShell for every system
```
Comment (TylerLeonhardt) [inline preview/snap/snapcraft.yaml]: nit: GitHub has an exclamation mark at the end. Not sure if you want to add that.

File: tools/releaseBuild/build.json [hunk 3]
```diff
@@ -12,13 +12,31 @@
...
+    {
+      "Name": "powershell-snap-preview-latest",
+      "RepoDestinationPath": "/PowerShell",
+      "BuildCommand": "/powershell-snap.sh -location _RepoDestinationPath_ -destination _DockerVolume_ -preview",
+      "AdditionalContextFiles" :[ "./tools/releaseBuild/Images/GenericLinuxFiles/powershell-snap.sh","./tools/releaseBuild/Images/GenericLinuxFiles/powershell-snap.ps1"],
+      "DockerFile": "./tools/releaseBuild/Images/microsoft_powershell_ubuntu16.04/Dockerfile",
+      "DockerImageName": "ps-snap-ubunutu-16-04",
+      "BinaryBucket": "release"
+  }
+]
```
Comment (TylerLeonhardt) [inline tools/releaseBuild/build.json]: nit: the indenting of this file is off.

File: tools/releaseBuild/Images/GenericLinuxFiles/powershell-snap.ps1 [hunk 4]
```diff
@@ -24,6 +32,8 @@ Push-Location
 try {
     Write-Verbose "snapcraft version $(snapcraft --version)" -Verbose    
     Set-Location $location
+    Set-location $directory
```
Comment (TylerLeonhardt) [inline tools/releaseBuild/Images/GenericLinuxFiles/powershell-snap.ps1]: nit: maybe just do `Set-Location "$location\$directory"` or `Set-Location "$location/$directory"`?

**Review by TylerLeonhardt — APPROVED** [review]
LGTM just some nitpicks you don't have to fix

File: appveyor.yml (hunk 1 above)
Comment (TravisEz13) [reply appveyor.yml]: removed

File: preview/snap/snapcraft.yaml (hunk 2 above)
Comment (TravisEz13) [reply preview/snap/snapcraft.yaml]: Added

File: tools/releaseBuild/build.json (hunk 3 above)
Comment (TravisEz13) [reply tools/releaseBuild/build.json]: fixed

File: tools/releaseBuild/Images/GenericLinuxFiles/powershell-snap.ps1 (hunk 4 above)
Comment (TravisEz13) [reply tools/releaseBuild/Images/GenericLinuxFiles/powershell-snap.ps1]: fixed

**Comment by TylerLeonhardt** [comment]
LGTM!

**Comment by TravisEz13** [comment]
Any changes to the RFC will be addressed later

**Pull request merged**